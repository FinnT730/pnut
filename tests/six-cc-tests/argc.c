/* args: abc def hij */
void putstring(char * s) {
  while (*s) {
    putchar(*s);
    s = s + 1;
  }
}

int main(int argc, char ** args) {
  int i = 0;
  putstring("Number of arguments: ");
  putchar(48 + argc);
  putchar('\n');
  putstring("Arguments:\n");
  while (i < argc) {
    putstring("argv[");
    putchar(48 + i);
    putstring("] = ");
    putstring(args[i]);
    putchar('\n');
    i = i + 1;
  }
}
