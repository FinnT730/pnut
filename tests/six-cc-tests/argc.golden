Number of arguments: 4
Arguments:
argv[0] = ./tests/six-cc-tests/argc.exe
argv[1] = abc
argv[2] = def
argv[3] = hij
